app/src/main/java/com/sun/moviedb/view/detail/DetailMovieFragment.kt

```diff
@@ -45,10 +55,34 @@ class DetailMovieFragment : ViewModelBaseFragment<DetailMovieViewModel,FragmentD
             handleBusinessException(it)
         })
     }
+
     private fun initAdapter(adapter: ViewpagerFragmentAdapter) {
         val fragments = ArrayList<Fragment>()
         val fragmentTrailer = TrailerFragment.newInstance()
+        fragmentTrailer.setOnTrailerClickListener {
+            extractYoutubeUrl(StringUtils.getUrlYoutube(it.key))
+        }
+        fragmentTrailer.setOnFirstTrailerListener {
+            extractYoutubeUrl(StringUtils.getUrlYoutube(it))
+        }
         fragments.add(fragmentTrailer)
         adapter.submitData(fragments)
     }
+    @SuppressLint("StaticFieldLeak")
+    private fun extractYoutubeUrl(url: String) {
+        val mExtractor = object : YouTubeExtractor(context!!) {
+            override fun onExtractionComplete(sparseArray: SparseArray<YtFile>?, videoMeta: VideoMeta) {
+                sparseArray?.let {
+                    playManager.release()
+                    playManager.init(context!!, playerViewTrailer, sparseArray.get(22).url)
```

Comment:
what is 22 ?

Reply:
Tại thư viện extract từ link youtube ra link có thể play được trả về sparseArray. thì possition 22 là link play video a ạ